tests/test_catalog.py

```diff
@@ -106,3 +108,128 @@ def test_fetch_catalog_initial_commit_for_empty_repo(
     assert r.repo.head
     assert r.working_tree_dir
     assert r.repo.head.commit.message == "Initial commit"
+
+
+def setup_catalog_for_push(config: Config, cluster: Cluster):
+    repo = catalog.fetch_catalog(config, cluster)
+    with open(repo.working_tree_dir / "test.txt", "w") as f:
+        f.writelines(["foo"])
+
+    repo.stage_files(["test.txt"])
+
+    return repo
+
+
+# Commodore doesn't support `catalog compile --push --local`, so we omit that test case.
```

Comment:
Is there a test to check this is indeed not possible?

Reply:
I realized that the logic preventing it is right there in `_push_catalog()` so I changed the test to check all combinations of `--local` and `--push`.